overlord/hookstate/hooks.go

```diff
@@ -71,16 +72,29 @@ func SetupPreRefreshHook(st *state.State, snapName string) *state.Task {
 	return task
 }
 
-func SetupGateAutoRefreshHook(st *state.State, snapName string, base, restart bool) *state.Task {
+type gateAutoRefreshAction int
+
+const (
```

Comment:
these probably merit a doc comment, also maybe they should be defined in snapstate itself?

Reply:
Agreed, I initially defined them here close to the (upcoming) hook handler, but snapstate is a better place (also in the light of potentially moving handler code there as discussed).

Reply:
Done, I had to rename the consts since there was a collision with ProceedWithRefresh function.